### ISSUE TITLE
Soft wrap long code output in pdf_output

### ISSUE THREAD

**Issue by ColinFay** [yihui/bookdown-crc#5]
Hey, 

First of all, thanks for this template, it saved us a lot of time :) 

I'm facing a little issue I can't figure out (and thanks @cderv from trying to solve it with me 💪): 

When printing the book to pdf, some of the code outputs overflow the width of the pdf. 

This happens with the default template from here, and can be reproduced as such: 

```bash
cd /tmp 
git clone https://github.com/yihui/bookdown-crc
cd bookdown-crc/
R -e 'bookdown::render_book("index.Rmd", "bookdown::pdf_book");browseURL(.Last.value)'
```

Here is the output of page 9:

![Screenshot 2020-11-19 at 09 39 02](https://user-images.githubusercontent.com/17936236/99642715-18fc4880-2a4c-11eb-87ba-11ddd88aaacb.png)

What I've tried:  

- Following https://bookdown.org/yihui/rmarkdown-cookbook/text-width.html: 
  - `options(width = 30)` 
  - listings in preamble.tex
```
\usepackage{listings}
\lstset{
  breaklines=true
}
```
- definie a knitr kook as in https://github.com/yihui/knitr-examples/blob/master/077-wrap-output.Rmd

- Change of the font size as defined in https://yihui.org/en/2018/08/bookdown-crc/

None of these seemed to solve the issue completely: changing the font size works to a certain extent, but for very long line the code still overflow.  

Are there any solution to correct this? 

Thanks for your help, 
Colin

**Comment by yihui** [comment]
This actually has been a mystery to me, because I don't know why the lines were automagically wrapped in my books and I didn't use listings or knitr hooks, e.g.,

![image](https://user-images.githubusercontent.com/163582/99978237-5114cb80-2d6b-11eb-976c-04d92f8a6c04.png)

I'll take a closer look, but can't spend too much time on it for now, so I can't promise I'll figure it out. You can proceed with writing your book, and I'm sure this problem is fixable.

**Comment by yihui** [comment]
Should be fixed with ad8521c, but I still don't understand why the lines could be automatically wrapped...
